### PR TITLE
Add Compressor docs

### PR DESCRIPTION
**Pull request by arturi**

### CODE REVIEW

File: docs/compressor.mdx [hunk 1]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
...
+
+The `@uppy/compressor` plugin optimizes images (JPEG, PNG) before upload, saving
+up to 60% in size (roughly 18 MB for 10 images). It uses
+[Compressor.js](https://github.com/fengyuanchen/compressorjs) library under the
+hood.
+
+## When should I use it?
+
+When your users are likely to upload images, potentially on mobile devices, and
+saving data / faster uploading speed is important.
```
Comment (aduh95) [inline docs/compressor.mdx:15]: The speed is the same, technically 🤓
```suggestion
saving data and faster uploads are important.
```

File: docs/compressor.mdx [hunk 2]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
...
+show an [Informer](/docs/user-interfaces/elements/informer) message with saved
+bytes, and then begin the upload as usual.
+
+## API
+
+### Options
+
+:::tip You can also pass any of the
+[Compressor.js options](https://github.com/fengyuanchen/compressorjs#options)
+here as well. :::
```
Comment (aduh95) [inline docs/compressor.mdx:68]:
```suggestion
:::tip

You can also pass any of the [Compressor.js options][] here as well.

:::
```

File: docs/compressor.mdx [hunk 3]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
 ---
 
 # Compressor
+
+The `@uppy/compressor` plugin optimizes images (JPEG, PNG) before upload, saving
+up to 60% in size (roughly 18 MB for 10 images). It uses
+[Compressor.js](https://github.com/fengyuanchen/compressorjs) library under the
```
Comment (aduh95) [inline docs/compressor.mdx:9]: Let's use a ref at the end
```suggestion
[Compressor.js][] library under the
```

File: docs/compressor.mdx [hunk 4]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
...
+	},
+};
+```
+
+## Events
+
+#### `compressor:complete`
+
+The event is emitted when all files are compressed. You can use it for side
+effects or custom UI notifications.
```
Comment (aduh95) [inline docs/compressor.mdx:111]:
```suggestion
effects or custom UI notifications.

[Compressor.js]: https://github.com/fengyuanchen/compressorjs
[Compressor.js options]: https://github.com/fengyuanchen/compressorjs#options
```

File: docs/compressor.mdx [hunk 5]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
...
+
+### `id`
+
+A unique identifier for this plugin (`string`, default: `'Compressor'`).
+
+### `quality`
+
+The quality of the output image passed to
+[Compressor.js](https://github.com/fengyuanchen/compressorjs) (`number`,
+default: `0.6`)
```
Comment (aduh95) [inline docs/compressor.mdx:78]:
```suggestion
[Compressor.js][] (`number`, default: `0.6`).
```

File: docs/compressor.mdx [hunk 6]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
...
+
+It must be a number between `0` and `1`. Be careful to use `1` as it may make
+the size of the output image become larger. In most cases, going with the
+default value is best.
+
+:::note This option is only available for `image/jpeg` and `image/webp` images.
+:::
```
Comment (aduh95) [inline docs/compressor.mdx:85]:
```suggestion
:::note

This option is only available for `image/jpeg` and `image/webp` images.

:::
```

File: docs/compressor.mdx [hunk 7]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
...
+
+### `limit`
+
+Number of images that will be compressed in parallel (`number`, default: `10`)
```
Comment (aduh95) [inline docs/compressor.mdx:89]:
```suggestion
Number of images that will be compressed in parallel (`number`, default: `10`).
```

File: docs/compressor.mdx [hunk 8]
```diff
@@ -3,3 +3,109 @@ sidebar_position: 11
 ---
 
 # Compressor
+
+The `@uppy/compressor` plugin optimizes images (JPEG, PNG) before upload, saving
```
Comment (aduh95) [inline docs/compressor.mdx:7]:
```suggestion
The `@uppy/compressor` plugin optimizes images (JPEG, PNG, and any other format supported by the client's browser) before upload, saving
```

Comment (arturi) [reply docs/compressor.mdx:7]: Are we sure about that?

**Pull request merged**